### PR TITLE
Definition generation on Requests and CSharp syntax formatting

### PR DESCRIPTION
**Pull request by galadril**
- Definition json generation on Requests
- CSharp / Json Syntax Formatting (and a copy button for easy copying)
- Copy buttons for Code / Definition

image

image

### CODE REVIEW

**Comment by galadril** [comment]
Hope you like the changes @cezarypiatek ;)
I was also planning to create formatter/styler for xml responses later (guess other PR)

**Comment by cezarypiatek** [comment]
I took your words for granted that it solves the issu

**Comment by galadril** [comment]
Ok thanks, so this PR is going to take you a bit more time ;)

**Comment by galadril** [comment]
Added also XML Syntax formatting and styling 
image

File: src/WireMockInspector/Views/CodeBlockViewer.cs [hunk 1]
```diff
@@ -1,141 +1,114 @@
...
+                    if (this.TextArea.TextView.LineTransformers.OfType<DiffLineBackgroundRenderer>().FirstOrDefault() is
+                        { } existing)
+                    {
+                        this.TextArea.TextView.LineTransformers.Remove(existing);
+                    }
+
+                    this.TextArea.TextView.LineTransformers.Add(new DiffLineBackgroundRenderer(md.oldTextLines));
+                    if (md.lang == "csharp")
+                    {
+                        this.TextArea.TextView.LineTransformers.Add(new CSharpSyntaxHighlighter());
```
Comment (cezarypiatek) [inline src/WireMockInspector/Views/CodeBlockViewer.cs:82]: Syntax highliting should be handled automatically by Avalonia.Edit and I'm pretty sure it was working correctly so far (for json and csharp). Why did you implemented custom syntax highlighers?

**Comment by cezarypiatek** [comment]
Ignore my last comment, I was checking it on mobile and I thought it was in different thread (previous pr). 
This PR is quite large and I need some time to review what actually was changed.....

File: src/WireMockInspector/Views/CodeBlockViewer.cs (hunk 1 above)
Comment (galadril) [reply src/WireMockInspector/Views/CodeBlockViewer.cs:82]: Actually it was just all white, i didnt saw any syntax highliting.

**Comment by cezarypiatek** [comment]
1) You mixed refactors with new features which makes it really hard to follow what actually was changed. Please avoid such practice in the future.
2) JsonCodeGenerator could be simply implemented by providing only a custom liquid template
3) Syntax highliting should my handled by Avalonia.Edit. If it doesn't work for you then probabaly something was broken during  dependency update made in previous PR.

I will try to fix those things later and integrated your changes.

**Comment by galadril** [comment]
ok thanks

btw JsonCodeGenerator does have a bit different code then the other CodeGenerator

**Comment by cezarypiatek** [comment]
Ok, you added a dedicated json escaping function and a default template selector. It could be incorporated in the original implementation. The only thing that I don't understand is the code below, could you explain what's the purpose od that replace?

> Regex.Replace(result, @",(?=\s*[\]}])", "");
